doc/api/process.md

```diff
@@ -146,20 +146,31 @@ most convenient for scripts).
 ### Event: 'uncaughtException'
 <!-- YAML
 added: v0.1.18
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/20097
+    description: Garbage collected unhandled rejections will now also trigger an
+                 uncaughtException. The `unhandledPromise` argument got added to
+                 distinguish those.
 -->
 
+* `err` {Error} The uncaught exception.
+* `unhandledPromise` {boolean} Indicates if the exception originates from an
+  unhandled rejection.
+
 The `'uncaughtException'` event is emitted when an uncaught JavaScript
 exception bubbles all the way back to the event loop. By default, Node.js
 handles such exceptions by printing the stack trace to `stderr` and exiting.
 Adding a handler for the `'uncaughtException'` event overrides this default
 behavior.
 
-The listener function is called with the `Error` object passed as the only
-argument.
-
 ```js
-process.on('uncaughtException', (err) => {
-  fs.writeSync(1, `Caught exception: ${err}\n`);
+process.on('uncaughtException', (err, unhandledPromise) => {
```

Comment:
I consider `process.on('uncaughtException')` a design mistake (I'm not alone in that, its original author does too) so I'd rather not touch it.